provisioning/baremetal_pod.go

```diff
@@ -1018,21 +1018,21 @@ func EnsureMetal3Deployment(info *ProvisioningInfo) (updated bool, err error) {
 	}
 
 	deploymentRolloutStartTime = time.Now()
-	deployment, updated, err := resourceapply.ApplyDeployment(info.Client.AppsV1(),
+	deployment, updated, err := resourceapply.ApplyDeployment(info.KubeClient.AppsV1(),
```

Comment:
So this resourceapply stuff only takes the native clients not the controller-runtime client. We need to either move to library-go client or change to  a different "apply" library..
https://github.com/kubernetes-sigs/cluster-api/blob/master/util/patch/patch.go#L83